linetools/analysis/continuumfnd.py

```diff
@@ -0,0 +1,206 @@
+              showcont=True, lam0=1500, dlam=20, yfmax=2, ymin=0,
+              outf=None, overwrite=False):
+    """ Finds continuum knots, and pixels that are located on continuum .
+        Identifies pixels that correspond to absorption lines, and do not includes them in continuum.
+
+        Parameters sm, npix, lchmin, ewsnlim that best define continuum might be different for different data sets.
+        Recomended using lt_continuumfit script (gui) after.
+
+    Parameters
+    ----------
+    spec : XSpectrum1D or str
+      input spectrum, or a file that contains input XSpectrum1D spectrum
+    sm : float
+      number of pixels how much to smooth input spectrum when defining continuum
+    npix : int
+      number of pixels, such that every maximum flux value is greater than for npix pixels with lower and at
+      higher wavelengths
+    lchmin : int
+      minimum number of pixels in the intervals that define continuum
+    ewsnlim : float
+      minimum value for S/N ratio of a quantity similar to equivalent width
+    showcont : bool
+      show continuum? if yes:
+      orange line with circles - continuum and knots
+      green - pixels used to define continuum knots
+      black - original spectrum , gray - error in flux
+      blue line - initially defined continuum (not used),
+                  and initially defined continuum shifted by the error in flux in the smoothed spectrum (also not used)
+      red line - smoothed spectrum
+    lam0 : float
+      lowest wavelength to plot
+    dlam : float
+      wavelenght interval width to plot; highest wavelength shown will be lam0+dlam
+    yfmax : float
+      maximum ylim is yfmax * np.median(flux)
+    ymin : float
+      minimum ylim is yfmin * np.median(flux)
+    outf : str
+      output file with knots (.jsn or .json type)
+    overwrite : bool
+      overwrite?
+
+    Returns:
+    --------
+    knots : list of 2-element lists
+      List of knots (wavelength, flux) that define continuum.
+    knotpixs : list of int
+      Indices of pixels that define continuum
+
+    """
+
+    # input spectrum
+    if type(spec) is xspec.XSpectrum1D:
+        sp = spec
+    else:
+        try:
+            sp = readspec(spec)
+        except OSError:
+            print('Not valid input. Please provide XSpectrum1D object, or a file that contains XSpectrum1D object')
+            return
+
+    wav = sp.wavelength
+    flx = sp.flux
+    sigf = sp.sig
+
+    # smoothed input spectrum by smsm pixels
+    sp3 = sp.box_smooth(sm)
+    wav3 = sp3.wavelength
+    flx3 = sp3.flux
+    sigf3 = sp3.sig
+
+
+    # 1) Define pixels that are on the continuum.
+    #    Smooth the spectrum, find pixels around maximums in the flux in the smoothed spectrum,
+    #    assume that the continuum is well defined by these pixels only.
+    #    This flux will be too high, and is used only in the step (1)
+    ipixs = []  # continuum pixels
+    for i in range(len(flx3)):
```

Comment:
Why a nested set of `for` loops?  
This is not Pythonic.  Is there another way?

Reply:
Now using one for loop and np.max. The code is also now faster